src/main/java/frc/robot/commands/Autos/WingOnly.java

```diff
@@ -54,86 +57,95 @@ public WingOnly(StateMachine subStateMachine, Climber subClimber, Drivetrain sub
-
-        Commands.waitSeconds(constShooter.AUTO_PREP_NONE_DELAY.in(Units.Seconds)),
+        Commands.runOnce(() -> subStateMachine.setTargetState(TargetState.PREP_VISION)),
+
+        // -- PRELOAD --
+        Commands.deferredProxy(() -> subStateMachine.tryState(RobotState.INTAKING, subStateMachine, subClimber,
+            subDrivetrain, subElevator, subIntake, subTransfer, subShooter))
+            .until(() -> subTransfer.getGamePieceCollected()),
+
+        Commands.waitUntil(() -> subShooter.readyToShoot()),
+
+        // Shoot! (Ends when we don't have a game piece anymore)
+        Commands.deferredProxy(() -> subStateMachine
+            .tryState(RobotState.SHOOTING, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake,
+                subTransfer,
+                subShooter)
+            .until(() -> !subTransfer.getGamePieceCollected())),
+
+        // Reset subsystems to chill
+        Commands.deferredProxy(() -> subStateMachine
+            .tryState(RobotState.NONE, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake, subTransfer,
+                subShooter)),
+
+        // -- W1 / W3 --
+        Commands.runOnce(() -> subStateMachine.setTargetState(TargetState.PREP_VISION)),
+
+        // Drive to first note (Intaking is within the path)
+        new PathPlannerAuto(determinePathName() + ".1"),
+
+        Commands.waitUntil(() -> subShooter.readyToShoot()),
+
+        // Shoot! (Ends when we don't have a game piece anymore)
+        Commands.deferredProxy(() -> subStateMachine
+            .tryState(RobotState.SHOOTING, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake,
+                subTransfer,
+                subShooter)
+            .until(() -> !subTransfer.getGamePieceCollected())),
+
+        // Reset subsystems to chill
+        Commands.deferredProxy(() -> subStateMachine
+            .tryState(RobotState.NONE, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake, subTransfer,
+                subShooter)),
+
+        // -- W2 --
+        Commands.runOnce(() -> subStateMachine.setTargetState(TargetState.PREP_VISION)),
+
+        // Drive to first note (Intaking is within the path)
+        new PathPlannerAuto(determinePathName() + ".2"),
+
+        Commands.waitUntil(() -> subShooter.readyToShoot()),
+
+        // Shoot! (Ends when we don't have a game piece anymore)
+        Commands.deferredProxy(() -> subStateMachine
+            .tryState(RobotState.SHOOTING, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake,
+                subTransfer,
+                subShooter)
+            .until(() -> !subTransfer.getGamePieceCollected())),
+
+        // Reset subsystems to chill
+        Commands.deferredProxy(() -> subStateMachine
+            .tryState(RobotState.NONE, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake, subTransfer,
+                subShooter)),
+
```

Comment:
This logic is repeated a lot (even repeated in preload and taxi) it might be smart to have a like command VisionDrivePathShoot that can be called multiple times with parameters because the steps are the same and this ensures we don't forget to do one.